### PR TITLE
Document few aspects of `.toMatchTypeOf` behaviour

### LINKED ISSUE #55: `.toMatchTypeOf()` does not take into account `readonly` properties (undocumented case?)
From https://github.com/mmkal/expect-type/issues/52#issuecomment-1987805501

The documentation suggests that `.toMatchTypeOf()` can be used to "allow for extra properties". Just wanted to draw attention that it might need better description and/or examples.

Currently at least for me it sounds like `.toMatchTypeOf()` and `.toEqualTypeOf()` are identical, but `.toMatchTypeOf()` allows testing partial match. That seems to be incorrect, when `readonly` gets involved:

```ts
import { expectTypeOf } from 'expect-type';

expectTypeOf<{ readonly a: string; b: number }>().not.toEqualTypeOf<{ a: string; b: number }>();

expectTypeOf<{ readonly a: string; b: number }>().not.toMatchTypeOf<{ a: string }>(); // should also pass, but it fails
```

[playground](https://www.typescriptlang.org/play?ts=5.3.3#code/JYWwDg9gTgLgBAbzgUwB5mQYxgFQJ4YDyAZnAL5zFQQhwDkaG2AtDAcnQNwBQ3jWudiQA8SKMgCGAEwgA7ADZ44EgFxwAzjCjBZAc05wARmtkBXEIeRRyAPgAUASgB0siDCcwIAUQCOpifL4RMSiymqa2noGxnBmFla2jjx86AJByCJikjIKSqoaWjr6RibmltZk9s6u7p4AshIwmAAW6ZlhBZG6iQ4GAPR9Gs0QpvJSyvLqEHBgEurqADRGpvDA8MQSwJPcQA)

### PR DESCRIPTION
**Pull request by mrazauskas**
From https://github.com/mmkal/expect-type/issues/55#issuecomment-2013750056

Closes #55
Closes #66 (perhaps?)

@mmkal Thanks for explanations. Here is an attempt to summarise of how I understood the intended behaviour of`.toMatchTypeOf`. Feel free to rework or to clarify further.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ import {expectTypeOf} from 'expect-type'
 import {foo, bar} from '../foo'
 
 // make sure `foo` has type {a: number}
-expectTypeOf(foo).toMatchTypeOf<{a: number}>()
+expectTypeOf(foo).toEqualTypeOf<{a: number}>()
```
Comment (mrazauskas) [inline README.md:15]: Looks like a typo.

File: README.md [hunk 2]
```diff
@@ -714,4 +737,4 @@ Once you're ready to make a pull request: clone the repo, and install pnpm if yo
...
 
-After adding the tests, run `npm run lint -- --fix` to update the readme, and `npm test -- --updateSnapshot` to update the errors test. The generated documentation and tests should be pushed to the same branch as the source code, and submitted as a pull request. CI will test that the docs and tests are up to date if you forget to run these commands.
+After adding the tests, run `npm run lint -- --fix` to update the readme, and `npm test -- --update` to update the errors test. The generated documentation and tests should be pushed to the same branch as the source code, and submitted as a pull request. CI will test that the docs and tests are up to date if you forget to run these commands.
```
Comment (mrazauskas) [inline README.md:740]: Might be this is outdated. `--updateSnapshot` did not work for me.

File: README.md [hunk 3]
```diff
@@ -145,12 +174,6 @@ expectTypeOf<never>().toBeNever()
 expectTypeOf<never>().toBeNumber()
 ```
 
-`.toEqualTypeOf` distinguishes between deeply-nested `any` and `unknown` properties:
```
Comment (mrazauskas) [inline README.md:148]: This is moved up.

File: README.md (hunk 2 above)
Comment (aryaemami59) [reply README.md:740]: Try `--update` or just `-u`

Comment (mrazauskas) [reply README.md:741]: That’s right. `--update` does work as expected, hence I change this text.

Comment (aryaemami59) [reply README.md:741]: Yeah I think when we moved to Vitest recently we forgot to update this section of the docs.

**Review by aryaemami59 — APPROVED** [review]
Looks good to me, if @mmkal gives the final ok we can go ahead and merge.

**Pull request closed without merging**